tests/e2e/tests/test_ping.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2022-2025, Chain4Travel AG. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package tests
+
+import (
+	"context"
+	"testing"
+
+	pingv1 "buf.build/gen/go/chain4travel/camino-messenger-protocol/protocolbuffers/go/cmp/services/ping/v1"
+	typesv1 "buf.build/gen/go/chain4travel/camino-messenger-protocol/protocolbuffers/go/cmp/types/v1"
+	"github.com/chain4travel/camino-messenger-bot/internal/metadata"
+	botGenerated "github.com/chain4travel/camino-messenger-bot/internal/rpc/generated"
+	"github.com/chain4travel/camino-messenger-bot/tests/e2e/bot"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func TestPingV1(t *testing.T, tt *Test) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	require.NoError(t, tt.caminoNetwork.Client.RegisterCMService(ctx, botGenerated.PingServiceV1))
+
+	supplierPartnerPlugin := tt.CreatePartnerPlugin(ctx, t, nil)
+
+	// bot with partnerPlugin and without rpc server (supplier)
+	supplierBot := tt.CreateBot(ctx, t, false, supplierPartnerPlugin, nil, []bot.CMService{
+		{Name: botGenerated.PingServiceV1, Fee: 100},
+	})
+
+	// bot without partnerPlugin and with rpc server (distributor)
+	distributorBot := tt.CreateBot(ctx, t, true, nil, nil, nil)
+
+	resp, err := distributorBot.PingServiceV1.Ping(
+		requestContext(ctx, &metadata.Metadata{
+			Recipient: supplierBot.CMAccountAddress().Hex(),
+		}),
+		&pingv1.PingRequest{
+			Header:      &typesv1.RequestHeader{BaseHeader: &typesv1.Header{}},
+			PingMessage: "ping",
+			Timestamp:   timestamppb.Now(),
+		},
+	)
+	require.NoError(t, err)
```

Comment:
I'm expecting also a check that the ping message has been mirrored. 
Only checking for success might be not enough.

So maybe it would be good to extend the ping message with something unique and check whether this is returned in the response.

Reply:
well, pp-mock is not mirroring:
```go
	return &pingv1.PingResponse{
		Header: &typesv1.ResponseHeader{
			Status: typesv1.StatusType_STATUS_TYPE_SUCCESS,
		},
		PingMessage: fmt.Sprintf("Ping response to [%s] with request ID: %s", request.PingMessage, md.RequestID),
	}, nil
```

Reply:
I can check for some string match using that message
Or I can change pp-mock behaviour

Reply:
Even better - so this is then a prime example why the test should test for the expected behavior which is in the case of the ping request to receive the message sent back.

The test should fail if the pp-mock doesn't do that correctly.